internal/js/modules/k6/browser/common/helpers.go

```diff
@@ -138,34 +130,27 @@ func createWaitForEventHandler(
 	ch := make(chan any)
 
 	go func() {
+		defer func() {
+			close(ch)
+			evCancelFn()
+		}()
 		for {
 			select {
 			case <-evCancelCtx.Done():
 				return
 			case ev := <-chEvHandler:
-				if stringSliceContains(events, ev.typ) {
+				if slices.Contains(events, ev.typ) {
 					if predicateFn != nil {
 						if predicateFn(ev.data) {
 							select {
 							case ch <- ev.data:
 							case <-evCancelCtx.Done():
-								return
 							}
-						}
-					} else {
-						select {
-						case ch <- nil:
-						case <-evCancelCtx.Done():
+							// We wait for one matching event only,
+							// then remove the event handler by cancelling context and stopping goroutine.
 							return
 						}
 					}
-					close(ch)
-
-					// We wait for one matching event only,
-					// then remove event handler by cancelling context and stopping goroutine.
-					evCancelFn()
-
-					return
```

Comment:
This is connected to what I wrote in https://github.com/grafana/k6/pull/4918/files#r2208193018.

We can receive more than one event of the same type, and the first one isn't necessarily the one we want.

Reply:
Actually, this changes the behaviour of `createWaitForEventHandler`. It now doesn't exit until the predicate function passes when it returns `true`. I think we should revert this change and use `createWaitForEventPredicateHandler`.